samples/drivers/adc/boards/stm32l496g_disco.conf

```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2021 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_LOG=y
```

Comment:
Why is a board overlay config needed for this?

Reply:
> agree with @MaureenHelm , if we need logging enabled, this should be done in the main config and not be board specific.

In stm32l496g_disco platform, by default vref is not enabled, so i have enabled it. If vref is not enabled, the ADC converted values will be incorrect and a user should be informed in such a case. This is specifically for this platform and test.

Reply:
@MaureenHelm & @nashif Can you help us to review and support on above details from Affrin?

Reply:
please move this into the main prj.conf, and avoid having an overlay just for logging on one platform. logging should be enabled globally here.

Reply:
@nashif Doing this will be enabled for all other platforms. Is it fine? 
@LMESTM @erwango Can you please suggest and support.

Reply:
@affrinpinhero-2356 CONFIG_LOG is already enabled in lot of samples, so this makes sense adding it in this one as well.
Alternatively, if the concern is that we need a way to inform user of potential issue in ADC initialization, then it should be done for all cases of ADC usage, not only in this sample.
So I'd suggest taking care in board.c directly, cf comment